src/components/ProgressBar/ProgressBar.unit.spec.jsx

```diff
@@ -12,5 +13,91 @@ describe('<ProgressBar/>', () => {
     tree.unmount();
   });
 
-  it('should pass aXe check', () => axeCheck(<ProgressBar percent={35} label="aria label here"/>));
+  it('should pass aXe check', () =>
+    axeCheck(<ProgressBar percent={35} label="aria label here" />));
+
+  describe('analytics event', function () {
+    it('should be triggered when a ProgressBar is mounted at 0%', () => {
+      const handleAnalyticsEvent = sinon.spy();
+
+      global.document.body.addEventListener(
+        'component-library-analytics',
+        handleAnalyticsEvent,
+      );
+
+      const tree = mount(
+        <ProgressBar percent={0} label="This is my ProgressBar." />,
+      );
+
+      global.document.body.removeEventListener(
+        'component-library-analytics',
+        handleAnalyticsEvent,
+      );
+
+      tree.unmount();
+
+      expect(
+        handleAnalyticsEvent.calledWith(
+          sinon.match.has('detail', {
+            componentName: 'ProgressBar',
+            action: 'change',
+            details: {
+              percent: 0,
+              label: 'This is my ProgressBar.',
+            },
+          }),
+        ),
+      ).to.be.true;
+    });
+
+    it('should be triggered when a ProgressBar updates to 100%', () => {
+      const handleAnalyticsEvent = sinon.spy();
+
+      global.document.body.addEventListener(
+        'component-library-analytics',
+        handleAnalyticsEvent,
+      );
+
+      const tree = mount(
+        <ProgressBar percent={42} label="This is my ProgressBar." />,
+      );
+
+      tree.setProps({ percent: 100 });
+
+      global.document.body.removeEventListener(
+        'component-library-analytics',
+        handleAnalyticsEvent,
+      );
```

Comment:
Why are we removing this event listener in the middle of a test shortly after adding it?

Reply:
Goood question.  This and the unmount have been dropped at the bottom of all existing tests.